src/internal/SetupResumeAcceptor.h

```diff
@@ -52,11 +55,16 @@ class SetupResumeAcceptor final {
       folly::exception_wrapper ex);
   void removeConnection(const std::shared_ptr<FrameTransport>& transport);
 
+  void closeAllConnections();
+
   std::shared_ptr<FrameSerializer> getOrAutodetectFrameSerializer(
       const folly::IOBuf& firstFrame);
 
-  std::unordered_set<std::shared_ptr<FrameTransport>> connections_;
+  std::shared_ptr<std::unordered_set<std::shared_ptr<FrameTransport>>> connections_{
```

Comment:
this doesn't need to be a `shared_ptr` and I think a `bool destroyed_` is clearer than nulling this out anyway.